### PR TITLE
Support simulation_context in scheduler

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves part of #6810 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/scheduler/job.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ class Job:
 
     def __init__(self, scheduler: Scheduler, real: Realization) -> None:
         self.real = real
+        self.state = State.WAITING
```
Comment (berland) [inline src/ert/scheduler/job.py:57]: This changes `job.py` from having the state implicitly to explicitly (!)

File: src/ert/scheduler/job.py [hunk 2]
```diff
@@ -141,6 +142,7 @@ async def _max_runtime_task(self) -> None:
         self.returncode.cancel()  # Triggers CancelledError
 
     async def _send(self, state: State) -> None:
+        self.state = state
```
Comment (berland) [inline src/ert/scheduler/job.py:145]: Maybe slightly naughty to have a state change as a bi-product of sending events. Unless we claim it is an implementation detail.

File: src/ert/scheduler/job.py (hunk 1 above)
Comment (berland) [reply src/ert/scheduler/job.py:57]: Only simulation_context (Everest) uses this.

File: src/ert/scheduler/scheduler.py [hunk 3]
```diff
@@ -110,9 +124,7 @@ def add_dispatch_information_to_jobs_file(self) -> None:
         for job in self._jobs.values():
             self._update_jobs_json(job.iens, job.real.run_arg.runpath)
 
-    async def execute(
-        self,
-    ) -> str:
+    async def execute(self, minimum_required_realizations: int = 0) -> str:
```
Comment (berland) [inline src/ert/scheduler/scheduler.py:127]: Supporting `minimum_required_realizations` is part of stop_long_running.

File: src/ert/scheduler/job.py (hunk 2 above)
Comment (pinkwah) [reply src/ert/scheduler/job.py:147]: Imo it's fine. We can rename the function to `_set_state` and have sending events be the side-effect but it's whatever.

**Review by pinkwah — APPROVED** [review]

**Pull request merged**